docs/getting-started-guides/scratch.md

```diff
@@ -0,0 +1,658 @@
+"virtual" IPs for pods and services and a different virtual ethernet device for each pod.
+
+There are broadly speaking two approaches to configuring in-cluster networking:
+- Create an Overlay network
+  - Easier to setup
+  - Traffic is encapsulated, so per-pod virtual IPs are routable.
+  - Examples:
+    - [Flannel](https://github.com/coreos/flannel) 
+    - [Weave](http://weave.works/)
+  - Does not require "Routes" portion of Cloud Provider module.
+- Configure network to route Pod IPs
+  - Harder to setup from scratch.
+  - Implemented for GCE and AWS.
+  - Uses Cloud Provider modules with "Routes" subinterface.
+  - Theoretically possible to implement for bare-metal networking, but has not been tried.
+    - Need to add routes to machines and/or routers for all Pod IPs.
+
+You need to select an address range for the virtual IPs. 
+- Various approaches:
+  - GCE: each project has own `10.0.0.0/8`.  Carve off a `/16` from that.  Room for several clusters in there.
+  - AWS: use one VPC for whole organization, carve off a chunk for each cluster.  Or use different VPC for different clusters.
+  - IPv6 not supported yet.
+- Allocate one CIDR for PodIPs for each node, or a large CIDR from which
+  smaller CIDRs are automatically allocated to each node (if nodes are dynamically added).
+  - You need Max-pods-per-node * max-number-of-nodes-expected IPs.
+  - e.g. use 10.240.0.0/16
+  - Need to make these routable or connect with overlay.
+- Allocate a subnet for services.  
+  - Does not need to be routable or known to overlay; kube-proxy will translate these to
+    PodIPs.
+  - Call this `SERVICE_CLUSTER_IP_RANGE`
+- Pick a static IP for master node.
+  - Call this the `MASTER_IP`
+- Open any firewalls which would hinder communication between the nodes.
+- Open any firewalls to allow access to the apiserver ports 80 and/or 443.
+
+### Other preparations
+
+#### Selecting a Cluster Name
+
+You should pick a name for your cluster.  Pick a short name for each cluster
+which is unique from future cluster names. This will be used in several ways:
+  - by kubectl to distinguish between various clusters you have access to.  You will probably want a
+    second one sometime later, such as for testing new Kubernetes releases, running in a different
+region of the world, etc.
+  - Kubernetes clusters can create cloud provider resources (e.g. AWS ELBs) and different clusters
+    need to distinguish which resources each created.  Call this `CLUSTERNAME`.
+
+#### Selecting a Security Model
+
+There are two main options for security:
+ 1. access the apiserver using HTTP
+   - use a firewall for security
+   - easier to setup
+ 1. access the apiserver using HTTPS  
+   - use https with certs, and credentials for user
+   - recommended approach
+   - configuring certs can be tricky
+
+If following the HTTPS approach, you will need to prepare certs and credentials.
+
+##### Preparing Certs
```

Comment:
@roberthbailey please review this section on certs for:
  - blatantly false statements
  - major missing information